### ISSUE TITLE
Documentation 

### ISSUE THREAD

**Issue by dufoli** [quarkiverse/quarkus-cxf#147]
Provide documentation in quarkiverse doc
Add link to it in readme. MD. 
Document process for client, request scoped, server process, add interceptor, security support, native build,...

**Comment by shumonsharif** [comment]
Nice!!!  I always wondered why there wasn't a Quarkiverse website ... glad to see there is one.  Happy to collaborate with you on this in a branch, let me know!

**Comment by dufoli** [comment]
https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/dev/index.html

**Comment by shumonsharif** [comment]
Hey @dufoli I don't believe Antora uses or depends on maven.  So we shouldn't actually be adding any pom.xml or module changes to parent pom.  I believe the way quarkiverse-docs hooks in and produces the website is via [this](https://github.com/quarkiverse/quarkiverse-docs/blob/master/antora-playbook.yml), which we'll need to submit a PR for.  Also, the branch I committed yesterday already has a docs folder with similar structure and the content we currently have on our existing README.md, feel free to commit directly into the branch while we finalize what the docs should look like.  I'm copying @gastaldi to confirm this.

**Comment by gastaldi** [comment]
That's right, the `docs` module will not be built on maven, but by Antora. That's the process we're adopting for now.

**Comment by dufoli** [comment]
I have edit my comment to reflect your comment. Thanks ;-)

**Comment by dufoli** [comment]
I have update description with structure. @shumonsharif  does it sounds ok ?

**Comment by shumonsharif** [comment]
Looks perfect, thanks @dufoli!!

**Comment by dufoli** [comment]
We have a lot to add later:
Code generation with maven plugin
Main method instead of container managed
A sentence about corba
Cient injection
Cxf Async
Wsdl
Literal, document, bare, raw mode
ServiceName and port name
WebServiceClient annotation
Link to cxf doc
Fastxml
Performance
...

**Comment by shumonsharif** [comment]
@dufoli that's a great list. I think many of these would be perfect candidates for an example codebase that demonstrates everything in action.

It seems the PR for #150 got closed for now, because we may need to bundle the codestarts as part of the extensions ... once that's available, we should be able to capture a lot of this as code examples.  Of course, we can document too ... what are your thoughts?

**Comment by omurthy** [comment]
You have added your own web server and client to the same project. I think they should be separate.

Can you add a sample WSDL-consuming tutorial? 
Like this: https://www.w3schools.com/xml/tempconvert.asmx?wsdl

**Comment by shumonsharif** [comment]
Hey @omurthy Regarding web server and client in the same project -- are you referring to the documentation, or the actual code?  Having server and client code in the **integration-tests** module for the extension is perfectly acceptable.  Let us know if there's some documentation that is off, as that's what this issue is about, otherwise feel free to open a new issue as well.

Regarding a WSDL-consuming tutorial, I've been exploring the idea of a **samples** project (similar to https://github.com/apache/cxf/tree/master/distribution/src/main/release/samples) that showcases all the various features of this extension and how to use them ... though not sure yet where the home for this will be.  @dufoli any thoughts?

**Comment by dufoli** [comment]
Intégration-tests and documentation seems enough. But we need to enrich them....

**Comment by omurthy** [comment]
The Server-side is clear but the Client-side is not enough I think.Maybe you should  start from here to enrich

**Comment by dufoli** [comment]
I see a lot of people complaining about documentation. But common it is an open source project. You can provide PR with improvment on documentation list : 
Code generation with maven plugin (wsdl 2 java) = example in integration-test
Main method instead of container managed DONE
A sentence about corba
Cient injection
Cxf Async
Wsdl
Literal, document, bare, raw mode
ServiceName and port name
WebServiceClient annotation
Link to cxf doc
Fastxml
Performance
Vertx transport
Interceptor producer for different sei
